game-core/src/main/java/games/strategy/triplea/delegate/battle/BattleState.java

```diff
@@ -28,13 +27,9 @@ public Side getOpposite() {
 
   Collection<Unit> getUnits(Side... sides);
 
-  Collection<Unit> getAttackingWaitingToDie();
+  Collection<Unit> getWaitingToDie(Side... sides);
 
-  Collection<Unit> getDefendingWaitingToDie();
-
-  Collection<Unit> getWaitingToDie(EnumSet<Side> sides);
-
-  void clearWaitingToDie(EnumSet<Side> sides);
+  void clearWaitingToDie(Side... sides);
```

Comment:
One would hope that battle state would take care of its own state, notably it would know when units should be cleared. This kind of API hints that this is more of a value object (or even a parameter) object, in which case having an interface at all is a bit overkill. No recommendation here, just noting the code smell here, that perhaps more logic needs to be moved into the class or it needs to have a pivot such that it's broken up and functionality injected into an imperative shell type of class.

Reply:
I'm leaning more towards battle state being a value object.  The battle steps will know the logic of how to manage the battle state.  The interface is just there to help define what the `BattleState` needs separately from `MustFightBattle`.

But, that might change once I've started breaking up the battleActions.